tst/EnergyPlus/unit/SurfaceGeometry.unit.cc

```diff
@@ -166,3 +166,67 @@ TEST_F( EnergyPlusFixture, BaseSurfaceRectangularTest )
 	EXPECT_FALSE( ErrorsFound );
 	EXPECT_EQ( Polygonal, Surface( ThisSurf ).Shape );
 }
+
+TEST_F( EnergyPlusFixture, ConfirmCheckSubSurfAzTiltNorm )
+{
+	SurfaceData BaseSurface;
+	SurfaceData SubSurface;
+	bool surfaceError;
+
+	//Case 1 - Base surface and subsurface face the same way - should be no error message and no surfaceError
+	surfaceError = false;
+	BaseSurface.Azimuth = 0.;
+	BaseSurface.Tilt = 0.;
+	BaseSurface.NewellSurfaceNormalVector.x = 0.;
+	BaseSurface.NewellSurfaceNormalVector.y = 0.;
+	BaseSurface.NewellSurfaceNormalVector.z = 1.;
+
+	SubSurface.Azimuth = 0.;
+	SubSurface.Tilt = 0.;
+	SubSurface.NewellSurfaceNormalVector.x = 0.;
+	SubSurface.NewellSurfaceNormalVector.y = 0.;
+	SubSurface.NewellSurfaceNormalVector.z = 1.;
+	checkSubSurfAzTiltNorm( BaseSurface, SubSurface, surfaceError );
+	EXPECT_FALSE( surfaceError );
+	EXPECT_TRUE( compare_err_stream( "" , true ) );
+
+	//Case 2 - Base surface and subsurface face the opposite way - should be error message and surfaceError=true
+	surfaceError = false;
+	SubSurface.Azimuth = 180.;
+	SubSurface.Tilt = 180.;
+	SubSurface.NewellSurfaceNormalVector.x = 1.;
+	SubSurface.NewellSurfaceNormalVector.y = 0.;
+	SubSurface.NewellSurfaceNormalVector.z = 0.;
+	checkSubSurfAzTiltNorm( BaseSurface, SubSurface, surfaceError );
+	EXPECT_TRUE( surfaceError );
+	//EXPECT_FALSE( compare_err_stream( "" , true ) );
```

Comment:
@mjwitte If you want to test whether there is any error message then you should [use has_err_output()](https://github.com/NREL/EnergyPlus/blob/331fff7b11eb72351ad034a59adddf0f205d3556/tst/EnergyPlus/unit/Fixtures/EnergyPlusFixture.hh#L236). One thing to keep in mind with with `has_err_output()` is that it will not clear/reset the text in the stream when called.

The EXPECTs in the `compare_*_stream()` are what drive the comparison since the `err_stream` variable is private in EnergyPlusFixture, so no they can not be removed.

I pulled down this branch and uncommented the various `compare_err_stream()` and I see that it is seg faulting on that EXPECT_EQ. That is curious and shouldn't be happening. I'm looking to see why.